pyiceberg/table/__init__.py

```diff
@@ -434,6 +460,8 @@ def overwrite(
         if table_arrow_schema != df.schema:
             df = df.cast(table_arrow_schema)
 
+        self.delete(delete_filter=overwrite_filter, snapshot_properties=snapshot_properties)
+
         with self.update_snapshot(snapshot_properties=snapshot_properties).overwrite() as update_snapshot:
```

Comment:
Does it make more sense to use self.update_snapshot(snapshot_properties=snapshot_properties).append() here since the overwrite already happens inside of self.delete()

Reply:
Thanks! This logic was still when we only supported the truncate operation, great suggestion, thanks 👍